applications/DG-Max/Algorithms/DGMaxDiscretization.cpp

```diff
@@ -92,8 +120,18 @@ void DGMaxDiscretization<DIM>::computeElementIntegrands(
                 elementMassMatrix(element, res);
                 return res;
             });
-        if (invertMassMatrix) {
-            massMatrix = massMatrix.inverse();
+        switch (massMatrixHandling) {
+            case DGMaxDiscretizationBase::NORMAL:
+                break;
+            case DGMaxDiscretizationBase::INVERT:
+                massMatrix = massMatrix.inverse();
+                break;
+            case DGMaxDiscretizationBase::ORTHOGONALIZE:
+                massMatrix.cholesky();
+                break;
+            default:
+                logger.assert_always(false,
+                                     "Not implemented mass matrix handling");
```

Comment:
I can see arguments for both approaches. A compiler warning might be caught earlier, but risks being overlooked. Then it won't be spotted till the next recompilation. 

The current version gives no warning, but at runtime it will always fail (assuming compiling in debug mode). The problem will thus be spotted late, but it will prevent a bug-hunt.

Any idea if there is a simple way to do both?

Reply:
the proper way would be to use the compiler warning and compile with `-Werror`, i.e. all warnings are errors, which would lead to a build break. This requires all warnings to be fixed though.